Assets/MixedRealityToolkit-SDK/Features/Utilities/Solvers/SolverHandler.cs

```diff
@@ -82,17 +142,63 @@ private void Update()
 
         private void LateUpdate()
         {
-            for (int i = 0; i < solvers.Count; ++i)
+            if (UpdateSolvers)
             {
-                Solver solver = solvers[i];
-
-                if (solver.enabled)
+                for (int i = 0; i < solvers.Count; ++i)
                 {
-                    solver.SolverUpdate();
+                    Solver solver = solvers[i];
+
+                    if (solver.enabled)
+                    {
+                        solver.SolverUpdate();
+                    }
                 }
             }
         }
 
+        protected void OnDestroy()
+        {
+            if (transformWithOffset != null)
+            {
+                Destroy(transformWithOffset);
```

Comment:
we should check if we're in the editor here and destroy immediate.

Reply:
What's the benefit? Unity recommends `Destroy` only for game code, which this is.

https://docs.unity3d.com/ScriptReference/Object.DestroyImmediate.html

>You are strongly recommended to use Destroy instead.

>In game code you should use Object.Destroy instead.

Reply:
In editor if you call Destroy, it'll throw warnings and suggest calling destroy immediate.

Reply:
Which is why checking if `Application.isEditor` before calling the appropriate method.

Reply:
This isn't editor code though. When will this be invoked in edit mode?